<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Sophia Müller-Dott" />

<meta name="date" content="2022-09-14" />

<title>Comparison methods</title>

<script src="site_libs/header-attrs-2.16/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">kinase_review</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Comparison methods</h1>
<h4 class="author">Sophia Müller-Dott</h4>
<h4 class="date">2022-09-14</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-09-14
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>kinase_review/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220620code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220620)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220620code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220620)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsaezlabkinasereviewtree752459d53f330f5daf10e275720df6368bf87d33targetblank752459da">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/saezlab/kinase_review/tree/752459d53f330f5daf10e275720df6368bf87d33" target="_blank">752459d</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsaezlabkinasereviewtree752459d53f330f5daf10e275720df6368bf87d33targetblank752459da"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/saezlab/kinase_review/tree/752459d53f330f5daf10e275720df6368bf87d33" target="_blank">752459d</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    INKA/
    Ignored:    code/.DS_Store
    Ignored:    code/IKAP/.DS_Store
    Ignored:    code/RoKAI-master/.DS_Store
    Ignored:    code/RoKAI-master/data/
    Ignored:    code/kinasePA/
    Ignored:    data/CPTAC_estefani_clean_zip/
    Ignored:    data/KEA3/
    Ignored:    output/KEA3/
    Ignored:    output/figures/

Untracked files:
    Untracked:  analysis/comparison_activity_estimation.Rmd
    Untracked:  code/KEA3/
    Untracked:  code/PTM-SEA/
    Untracked:  code/RoKAI/
    Untracked:  code/tmp_directPA.R

Unstaged changes:
    Modified:   code/IKAP/IKAP_input_CPTAC_ccrcc.csv
    Modified:   code/IKAP/IKAP_input_CPTAC_luad.csv
    Modified:   code/IKAP/IKAP_input_CPTAC_ucec.csv
    Modified:   code/IKAP/kinase_act_ccrcc.csv
    Modified:   code/IKAP/kinase_act_luad.csv
    Modified:   code/IKAP/kinase_act_ucec.csv
    Modified:   code/IKAP/kinase_ccrcc.csv
    Modified:   code/IKAP/kinase_luad.csv
    Modified:   code/IKAP/kinase_ucec.csv
    Modified:   code/IKAP/phosphositeplus.csv
    Modified:   code/IKAP/run_IKAP.m
    Deleted:    code/run_KSEA.R
    Deleted:    code/run_PTM-SEA.R
    Deleted:    code/src/io.R
    Deleted:    code/src/parse_yaml_ssgsea.R
    Deleted:    code/src/ssGSEA2.0.R
    Deleted:    code/src/utils.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(cmapR)</code></pre>
<pre><code>Warning: package &#39;cmapR&#39; was built under R version 4.1.1</code></pre>
<pre class="r"><code>library(org.Hs.eg.db)</code></pre>
<pre><code>Loading required package: AnnotationDbi</code></pre>
<pre><code>Warning: package &#39;AnnotationDbi&#39; was built under R version 4.1.2</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>Warning: package &#39;BiocGenerics&#39; was built under R version 4.1.1</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, append, as.data.frame, basename, cbind, colnames,
    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
    union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>Loading required package: Biobase</code></pre>
<pre><code>Warning: package &#39;Biobase&#39; was built under R version 4.1.1</code></pre>
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    &#39;browseVignettes()&#39;. To cite Bioconductor, see
    &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>Warning: package &#39;IRanges&#39; was built under R version 4.1.1</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>Warning: package &#39;S4Vectors&#39; was built under R version 4.1.3</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    expand.grid, I, unname</code></pre>
<pre><code></code></pre>
<pre class="r"><code>library(KSEAapp)
library(tidyverse)</code></pre>
<pre><code>Warning: package &#39;tidyverse&#39; was built under R version 4.1.2</code></pre>
<pre><code>── Attaching packages
───────────────────────────────────────
tidyverse 1.3.2 ──</code></pre>
<pre><code>✔ ggplot2 3.3.6     ✔ purrr   0.3.4
✔ tibble  3.1.8     ✔ dplyr   1.0.9
✔ tidyr   1.2.0     ✔ stringr 1.4.1
✔ readr   2.1.2     ✔ forcats 0.5.2</code></pre>
<pre><code>Warning: package &#39;ggplot2&#39; was built under R version 4.1.2</code></pre>
<pre><code>Warning: package &#39;tibble&#39; was built under R version 4.1.2</code></pre>
<pre><code>Warning: package &#39;tidyr&#39; was built under R version 4.1.2</code></pre>
<pre><code>Warning: package &#39;readr&#39; was built under R version 4.1.2</code></pre>
<pre><code>Warning: package &#39;dplyr&#39; was built under R version 4.1.2</code></pre>
<pre><code>Warning: package &#39;forcats&#39; was built under R version 4.1.2</code></pre>
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::collapse()   masks IRanges::collapse()
✖ dplyr::combine()    masks Biobase::combine(), BiocGenerics::combine()
✖ dplyr::desc()       masks IRanges::desc()
✖ tidyr::expand()     masks S4Vectors::expand()
✖ dplyr::filter()     masks stats::filter()
✖ dplyr::first()      masks S4Vectors::first()
✖ dplyr::lag()        masks stats::lag()
✖ ggplot2::Position() masks BiocGenerics::Position(), base::Position()
✖ purrr::reduce()     masks IRanges::reduce()
✖ dplyr::rename()     masks S4Vectors::rename()
✖ dplyr::select()     masks AnnotationDbi::select()
✖ dplyr::slice()      masks IRanges::slice()</code></pre>
<pre class="r"><code>source(&quot;code/method_functions.R&quot;)</code></pre>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Here we will compare the kinase actvities of each method. From the
methods in the review these will be considered in the compatison:</p>
<p>Considered: - IKAP - KARP - KSEA App - PTM-SEA - RoKAI - KEA3 (only
for Venn diagramm)</p>
<p>Not considered: - CLUE: time-series - KinasePA: multi-condition
studies - INKA: code only available for MaxQuant files</p>
<p>We tried to run the methods with their default settings. The methods
were run for three CPTAC studies (ccrcc, luad, ucec)</p>
</div>
<div id="kinase-activity-estimation" class="section level2">
<h2>Kinase activity estimation</h2>
<p>For the methods we used comparison between tumor samples and normal
samples from three CPTAC studies (ccrcc, luad, ucec).</p>
<pre class="r"><code>## Load data ---------------------------
CPTAC_data &lt;- list(ccrcc = read.csv(&quot;data/CPTAC_estefani_clean_zip/ccrcc/ccrcc_phospho_tvalues.csv&quot;),
                   luad = read.csv(&quot;data/CPTAC_estefani_clean_zip/luad/luad_phospho_tvalues.csv&quot;),
                   ucec = read.csv(&quot;data/CPTAC_estefani_clean_zip/ucec/ucec_phospho_tvalues.csv&quot;))

CPTAC_data &lt;- map(CPTAC_data, function(CPTAC_cohort){
  cbind(protein_symbol =  map_chr(str_split(CPTAC_cohort$ID, &quot;_&quot;), function(x) x[1]),
        pps =  map_chr(str_split(CPTAC_cohort$ID, &quot;_&quot;), function(x) x[2]),
        CPTAC_cohort)
})

## Add uniprot ID, filter NAs in t-values, filter peptides with multipls pps
CPTAC_data_filtered &lt;- map(CPTAC_data, function(CPTAC_cohort){
  mapped &lt;- mapIds(org.Hs.eg.db,
                   keys = CPTAC_cohort$protein_symbol,
                   keytype = &quot;SYMBOL&quot;,
                   column = c(&quot;UNIPROT&quot;))

  CPTAC_cohort &lt;- CPTAC_cohort %&gt;%
    add_column(protein_uniprot = mapped, .after = &quot;protein_symbol&quot;) %&gt;%
    filter(rowSums(is.na(.)) == 0) %&gt;%
    filter(str_count(pps, &quot;[A-Z]&quot;) == 1)

  CPTAC_cohort %&gt;% add_column(uniprot_ID = paste0(CPTAC_cohort$protein_uniprot, &quot;_&quot;, CPTAC_cohort$pps), .after = &quot;ID&quot;)

})</code></pre>
<pre><code>&#39;select()&#39; returned 1:many mapping between keys and columns
&#39;select()&#39; returned 1:many mapping between keys and columns
&#39;select()&#39; returned 1:many mapping between keys and columns</code></pre>
<div id="ikap" class="section level3">
<h3>IKAP</h3>
<p>t values input preparation</p>
<pre class="r"><code># Network
PPS.plus &lt;- read_tsv(file = &quot;data/PPSp_20220830&quot;, skip = 2)</code></pre>
<pre><code>Rows: 22291 Columns: 16
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (14): GENE, KINASE, KIN_ACC_ID, KIN_ORGANISM, SUBSTRATE, SUB_ACC_ID, SUB...
dbl  (2): SUB_GENE_ID, SITE_GRP_ID

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>IKAP.network &lt;- PPS.plus %&gt;% 
  dplyr::filter(KIN_ORGANISM == &quot;human&quot; &amp; SUB_ORGANISM == &quot;human&quot;) %&gt;%
  dplyr::select(KINASE, SUB_GENE, SUB_MOD_RSD) %&gt;%
  dplyr::mutate(PSP = paste(SUB_GENE, SUB_MOD_RSD, sep = &quot;_&quot;)) %&gt;%
  dplyr::rename(&quot;GENE&quot; = &quot;KINASE&quot;) %&gt;%
  dplyr::select(GENE, PSP)

# Samples
map(names(CPTAC_data_filtered), function(cohort){
 cohort_data &lt;- CPTAC_data_filtered[[cohort]]
 
  IKAP_data &lt;- data.frame(str_split_fixed(cohort_data$ID, &quot;_&quot;, 2)[,1]) %&gt;%
    add_column(peptide =cohort_data$ID,
      logFC =cohort_data$NATvsTUM_t)

  colnames(IKAP_data)[1] &lt;- &quot;protein&quot;
  write.csv(IKAP_data, paste0(&quot;code/IKAP/IKAP_input_CPTAC_&quot;,cohort, &quot;.csv&quot;), row.names = F)
})</code></pre>
<pre><code>[[1]]
NULL

[[2]]
NULL

[[3]]
NULL</code></pre>
<pre class="r"><code>write.csv(IKAP.network, paste0(&quot;code/IKAP/phosphositeplus.csv&quot;), row.names = F)</code></pre>
<p>IKAP was run in MATLAB and the results were processed in R</p>
<pre class="r"><code>IKAP_act_files &lt;- list.files(&quot;code/IKAP&quot;, pattern = &quot;kinase&quot;, full.names = T)
IKAP_names &lt;- str_remove(str_remove(IKAP_act_files, &quot;code/IKAP/kinase_act_&quot;), &quot;.csv&quot;)
IKAP_names &lt;- str_remove(IKAP_names, &quot;code/IKAP/kinase_&quot;)

IKAP_act &lt;- map(unique(IKAP_names), function(cohort){
  files &lt;- list.files(&quot;code/IKAP&quot;, pattern = cohort, full.names = T)
  act_df &lt;- cbind(read_csv(files[3], col_names = F), read_csv(files[2], col_names = F))
  names(act_df) &lt;- c(&quot;Kinase&quot;, &quot;Score&quot;)
  act_df %&gt;%
    arrange(desc(abs(Score)))
})</code></pre>
<pre><code>Rows: 246 Columns: 1
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (1): X1

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 246 Columns: 1
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
dbl (1): X1

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 249 Columns: 1
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (1): X1

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 249 Columns: 1
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
dbl (1): X1

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 242 Columns: 1
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (1): X1

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 242 Columns: 1
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
dbl (1): X1

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>names(IKAP_act) &lt;- unique(IKAP_names)

head(IKAP_act$ccrcc)</code></pre>
<pre><code>  Kinase Score
1   SGK3    15
2  CAMK4    15
3   JNK3    15
4  IRAK4    15
5    FAK    15
6   PYK2    15</code></pre>
</div>
<div id="karp" class="section level3">
<h3>KARP</h3>
<p>t values input preparation</p>
<pre class="r"><code># Network
PPS.plus.KARP &lt;- read_tsv(file = &quot;data/PPSp_20220830&quot;, skip = 2)  %&gt;% 
  dplyr::filter(KIN_ORGANISM == &quot;human&quot; &amp; SUB_ORGANISM == &quot;human&quot;) %&gt;%
  dplyr::select(KINASE, SUB_GENE, SUB_MOD_RSD) %&gt;%
  dplyr::mutate(TARGET = paste(SUB_GENE, SUB_MOD_RSD, sep = &quot;_&quot;)) %&gt;%
  dplyr::rename(&quot;GENE&quot; = &quot;KINASE&quot;) %&gt;%
  dplyr::select(GENE, TARGET)</code></pre>
<pre><code>Rows: 22291 Columns: 16
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (14): GENE, KINASE, KIN_ACC_ID, KIN_ORGANISM, SUBSTRATE, SUB_ACC_ID, SUB...
dbl  (2): SUB_GENE_ID, SITE_GRP_ID

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code># Samples
KARP.input &lt;- map(names(CPTAC_data_filtered), function(cohort){
 cohort_data &lt;- CPTAC_data_filtered[[cohort]]
 
 cohort_data &lt;- cohort_data %&gt;% 
   dplyr::select(ID, NATvsTUM_t)
})</code></pre>
<pre class="r"><code>KARP_act &lt;- map(KARP.input, function(sample){
  act_df &lt;- runKARP(t.values = sample, PKN = PPS.plus.KARP, min.targets = 1)
  names(act_df) &lt;- c(&quot;Kinase&quot;, &quot;Score&quot;)
  act_df %&gt;%
    arrange(desc(abs(Score)))})

names(KARP_act) &lt;- names(CPTAC_data_filtered)

head(KARP_act$ccrcc)</code></pre>
<pre><code>  Kinase    Score
1   CDK2 6516.691
2   CDK1 6048.864
3  PKACA 4133.828
4   ERK2 3321.189
5   ERK1 3066.544
6   Akt1 2254.254</code></pre>
</div>
<div id="ksea-app" class="section level3">
<h3>KSEA App</h3>
<p>t values input preparation</p>
<pre class="r"><code>KSEA.input &lt;- map(names(CPTAC_data_filtered), function(cohort){
 cohort_data &lt;- CPTAC_data_filtered[[cohort]]
  KSEA_data &lt;- data.frame(str_split_fixed(cohort_data$ID, &quot;_&quot;, 2)) %&gt;%
    tibble::add_column(p =cohort_data$NATvsTUM_adj.P.Val,
               FC = (2^cohort_data$NATvsTUM_t),
               Protein = str_split_fixed(cohort_data$uniprot_ID, &quot;_&quot;, 2)[,1],
               Peptide = &quot;NULL&quot;) %&gt;%
    dplyr::rename(&quot;Gene&quot; = &quot;X1&quot;, &quot;Residue.Both&quot; = &quot;X2&quot;)

  KSEA_data &lt;- KSEA_data[c(5,1,6,2,3,4)]
})</code></pre>
<pre class="r"><code>KSEA_act &lt;- map(KSEA.input, function(sample){
  KSEA.network &lt;- KSEAapp::KSData

  KSEA_scores &lt;- KSEA.Scores(KSData = KSEA.network,
                             PX = sample,
                             NetworKIN = T,
                             NetworKIN.cutoff = 5)

  KSEA_scores %&gt;%
    dplyr::select(Kinase.Gene, z.score) %&gt;%
    dplyr::rename(c(&quot;Kinase&quot; = &quot;Kinase.Gene&quot;, &quot;Score&quot; = &quot;z.score&quot;)) %&gt;%
    arrange(desc(abs(Score)))
    
})

names(KSEA_act) &lt;- names(CPTAC_data_filtered)

head(KSEA_act$ccrcc)</code></pre>
<pre><code>    Kinase     Score
1     CDK2  5.157705
2 MAPKAPK5 -5.064564
3     CDK1  4.704382
4      ATM  4.074623
5    HIPK1 -3.938763
6    PRKCZ -3.624710</code></pre>
</div>
<div id="ptm-sea" class="section level3">
<h3>PTM-SEA</h3>
<p>t values input preparation</p>
<pre class="r"><code>map(names(CPTAC_data_filtered), function(cohort){
 cohort_data &lt;- CPTAC_data_filtered[[cohort]]
 cohort_data &lt;- cohort_data %&gt;%
    dplyr::mutate(rid = paste0(str_replace(uniprot_ID, &quot;_&quot;, &quot;;&quot;), &quot;-p&quot;)) %&gt;%
    dplyr::select(c(&quot;rid&quot;, &quot;NATvsTUM_t&quot;)) %&gt;%
    dplyr::rename(&quot;t&quot; = &quot;NATvsTUM_t&quot;) %&gt;%
    column_to_rownames(&quot;rid&quot;)
 cohort_data_m &lt;- as.matrix(cohort_data)

 cohort_data_gct &lt;- new(&quot;GCT&quot;, mat=cohort_data_m)
 write_gct(cohort_data_gct, paste0(&quot;code/PTM-SEA/PTM-SEA_input_&quot;,cohort))
})</code></pre>
<pre><code>Saving file to  code/PTM-SEA/PTM-SEA_input_ccrcc_n1x33625.gct 
Dimensions of matrix: [33625x1]
Setting precision to 4
Saved.
Saving file to  code/PTM-SEA/PTM-SEA_input_luad_n1x38398.gct 
Dimensions of matrix: [38398x1]
Setting precision to 4
Saved.
Saving file to  code/PTM-SEA/PTM-SEA_input_ucec_n1x30710.gct 
Dimensions of matrix: [30710x1]
Setting precision to 4
Saved.</code></pre>
<pre><code>[[1]]
NULL

[[2]]
NULL

[[3]]
NULL</code></pre>
<pre class="r"><code>PTM_act_files &lt;- list.files(&quot;code/PTM-SEA/2022-09-14&quot;, pattern = &quot;ssGSEA-scores.gct&quot;, full.names = T, recursive = TRUE)
PTM_names &lt;- map_chr(str_split(PTM_act_files, &quot;_&quot;), function(x) x[3])
PTM_act &lt;- map(PTM_act_files, function(x){
  act &lt;- read.delim(file = x, skip=2)
  PTM &lt;- act[str_detect(act$id, &quot;KINASE&quot;), c(&quot;id&quot;, &quot;t&quot;)] %&gt;%
    dplyr::rename(c(&quot;Kinase&quot; = &quot;id&quot;, &quot;Score&quot; = &quot;t&quot;))
  PTM$Kinase &lt;- map_chr(str_split(PTM$Kinase, &quot;_&quot;), function(x) x[2])
  PTM$Kinase &lt;-  map_chr(str_split(PTM$Kinase, &quot;/&quot;), function(x) x[length(x)])
  PTM %&gt;%
    arrange(desc(abs(Score)))
})
names(PTM_act) &lt;- PTM_names

head(PTM_act$ccrcc)</code></pre>
<pre><code>  Kinase  Score
1   CDK1 6.8067
2  MAPK3 6.6309
3   CDK2 6.3862
4  MAPK1 5.4555
5  GSK3B 4.4400
6   AKT1 3.7144</code></pre>
</div>
<div id="rokai" class="section level3">
<h3>RoKAI</h3>
<p>t value input preparation</p>
<pre class="r"><code>map(names(CPTAC_data_filtered), function(cohort){
 cohort_data &lt;- CPTAC_data_filtered[[cohort]]
  rokai_data &lt;- data.frame(str_split_fixed(cohort_data$uniprot_ID, &quot;_&quot;, 2)) %&gt;%
    add_column(Quantification = cohort_data$NATvsTUM_t)

  colnames(rokai_data) &lt;- c(&quot;Protein&quot;, &quot;Position&quot;, &quot;Quantification&quot;)

  write.csv(rokai_data, paste0(&quot;code/RoKAI/rokai_input_&quot;,cohort, &quot;.csv&quot;))
})</code></pre>
<pre><code>[[1]]
NULL

[[2]]
NULL

[[3]]
NULL</code></pre>
<p>RoKAI kinase activity estimation was run using the web application <a
href="https://rokai.io/" class="uri">https://rokai.io/</a></p>
<pre class="r"><code>rokai_act_files &lt;- list.files(&quot;code/rokai&quot;, pattern = &quot;kinase_table&quot;, full.names = T)
rokai_names &lt;- str_remove(str_remove(rokai_act_files, &quot;code/rokai/kinase_table_&quot;), &quot;.csv&quot;)
rokai_act &lt;- map(rokai_act_files, function(file){
  res &lt;- read_csv(file)
  res %&gt;% 
    dplyr::select(Gene, ZScore) %&gt;%
    dplyr::rename(c(&quot;Kinase&quot; = &quot;Gene&quot;, &quot;Score&quot; = ZScore)) %&gt;%
    arrange(desc(abs(Score)))
  }
  )</code></pre>
<pre><code>Rows: 196 Columns: 9
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (3): KinaseID, Name, Gene
dbl (6): NumSubs, Activity, StdErr, ZScore, PValue, FDR

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 205 Columns: 9
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (3): KinaseID, Name, Gene
dbl (6): NumSubs, Activity, StdErr, ZScore, PValue, FDR

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 202 Columns: 9
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (3): KinaseID, Name, Gene
dbl (6): NumSubs, Activity, StdErr, ZScore, PValue, FDR

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>names(rokai_act) &lt;- rokai_names

head(rokai_act$ccrcc)</code></pre>
<pre><code># A tibble: 6 × 2
  Kinase  Score
  &lt;chr&gt;   &lt;dbl&gt;
1 MAPK3    4.30
2 CDK1     3.82
3 MAPK7    3.73
4 HIPK1   -3.71
5 RPS6KB1  3.54
6 PAK2     3.39</code></pre>
</div>
<div id="kea3" class="section level3">
<h3>KEA3</h3>
<pre class="r"><code>map(names(CPTAC_data_filtered), function(cohort){
 cohort_data &lt;- CPTAC_data_filtered[[cohort]]

  KEA3_data &lt;-cohort_data %&gt;%
    filter(NATvsTUM_adj.P.Val &lt;= 0.05) %&gt;%
    filter(abs(NATvsTUM_logFC) &gt;= 2)

  KEA3_data &lt;- unique(KEA3_data$protein_symbol)

  write.table(KEA3_data,  paste0(&quot;code/KEA3/KEA3_input_&quot;, cohort, &quot;.txt&quot;), row.names = F, col.names = F,  quote = FALSE)

})</code></pre>
<pre><code>[[1]]
NULL

[[2]]
NULL

[[3]]
NULL</code></pre>
<p>We also used the web application of KEA3 (<a
href="https://maayanlab.cloud/kea3/"
class="uri">https://maayanlab.cloud/kea3/</a>)</p>
<pre class="r"><code>KEA3_act_files &lt;- list.files(&quot;code/KEA3&quot;, pattern = &quot;mean_rank&quot;, full.names = T)
KEA3_names &lt;- str_remove(str_remove(KEA3_act_files, &quot;code/KEA3/mean_rank_&quot;), &quot;.tsv&quot;)
KEA3_act &lt;- map(KEA3_act_files, function(file){
  res &lt;- read_tsv(file)
  res$Protein &lt;- str_remove(res$Protein, &quot;\\*&quot;)
  res %&gt;% 
    dplyr::select(Rank, Protein) %&gt;%
    dplyr::rename(&quot;Kinase&quot; = &quot;Protein&quot;, &quot;Score&quot; = &quot;Rank&quot;)
})</code></pre>
<pre><code>Rows: 518 Columns: 4
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (2): Protein, Overlapping Proteins
dbl (2): Rank, Mean rank

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 519 Columns: 4
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (2): Protein, Overlapping Proteins
dbl (2): Rank, Mean rank

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 519 Columns: 4
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (2): Protein, Overlapping Proteins
dbl (2): Rank, Mean rank

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>names(KEA3_act) &lt;- KEA3_names

head(KEA3_act$ccrcc)</code></pre>
<pre><code># A tibble: 6 × 2
  Score Kinase
  &lt;dbl&gt; &lt;chr&gt; 
1     1 CASK  
2     2 EGFR  
3     3 SRC   
4     4 FYN   
5     5 PDGFRB
6     6 PRKCA </code></pre>
</div>
</div>
<div id="comparison" class="section level2">
<h2>Comparison</h2>
<div id="correlation" class="section level3">
<h3>Correlation</h3>
</div>
<div id="top-kinases" class="section level3">
<h3>Top kinases</h3>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] forcats_0.5.2        stringr_1.4.1        dplyr_1.0.9         
 [4] purrr_0.3.4          readr_2.1.2          tidyr_1.2.0         
 [7] tibble_3.1.8         ggplot2_3.3.6        tidyverse_1.3.2     
[10] KSEAapp_0.99.0       org.Hs.eg.db_3.14.0  AnnotationDbi_1.56.2
[13] IRanges_2.28.0       S4Vectors_0.32.4     Biobase_2.54.0      
[16] BiocGenerics_0.40.0  cmapR_1.6.0          workflowr_1.7.0     

loaded via a namespace (and not attached):
  [1] googledrive_2.0.0           colorspace_2.0-3           
  [3] ellipsis_0.3.2              rprojroot_2.0.3            
  [5] cytolib_2.6.2               XVector_0.34.0             
  [7] GenomicRanges_1.46.1        fs_1.5.2                   
  [9] rstudioapi_0.14             bit64_4.0.5                
 [11] fansi_1.0.3                 lubridate_1.8.0            
 [13] xml2_1.3.3                  cachem_1.0.6               
 [15] knitr_1.40                  jsonlite_1.8.0             
 [17] broom_1.0.0                 dbplyr_2.2.1               
 [19] png_0.1-7                   compiler_4.1.0             
 [21] httr_1.4.4                  backports_1.4.1            
 [23] assertthat_0.2.1            Matrix_1.4-1               
 [25] fastmap_1.1.0               gargle_1.2.0               
 [27] cli_3.3.0                   later_1.3.0                
 [29] htmltools_0.5.3             tools_4.1.0                
 [31] gtable_0.3.0                glue_1.6.2                 
 [33] GenomeInfoDbData_1.2.7      Rcpp_1.0.9                 
 [35] cellranger_1.1.0            jquerylib_0.1.4            
 [37] vctrs_0.4.1                 Biostrings_2.62.0          
 [39] xfun_0.32                   ps_1.7.1                   
 [41] rvest_1.0.3                 lifecycle_1.0.1            
 [43] gtools_3.9.3                googlesheets4_1.0.1        
 [45] getPass_0.2-2               zlibbioc_1.40.0            
 [47] scales_1.2.1                vroom_1.5.7                
 [49] RProtoBufLib_2.6.0          hms_1.1.2                  
 [51] promises_1.2.0.1            MatrixGenerics_1.6.0       
 [53] parallel_4.1.0              SummarizedExperiment_1.24.0
 [55] yaml_2.3.5                  memoise_2.0.1              
 [57] sass_0.4.2                  stringi_1.7.8              
 [59] RSQLite_2.2.16              flowCore_2.6.0             
 [61] caTools_1.18.2              GenomeInfoDb_1.30.1        
 [63] rlang_1.0.4                 pkgconfig_2.0.3            
 [65] matrixStats_0.62.0          bitops_1.0-7               
 [67] evaluate_0.16               lattice_0.20-45            
 [69] bit_4.0.4                   processx_3.7.0             
 [71] tidyselect_1.1.2            magrittr_2.0.3             
 [73] R6_2.5.1                    gplots_3.1.3               
 [75] generics_0.1.3              DelayedArray_0.20.0        
 [77] DBI_1.1.3                   withr_2.5.0                
 [79] pillar_1.8.1                haven_2.5.1                
 [81] whisker_0.4                 KEGGREST_1.34.0            
 [83] RCurl_1.98-1.8              modelr_0.1.9               
 [85] crayon_1.5.1                KernSmooth_2.23-20         
 [87] utf8_1.2.2                  tzdb_0.3.0                 
 [89] rmarkdown_2.16              grid_4.1.0                 
 [91] readxl_1.4.1                blob_1.2.3                 
 [93] callr_3.7.2                 git2r_0.30.1               
 [95] reprex_2.0.2                digest_0.6.29              
 [97] httpuv_1.6.5                RcppParallel_5.1.5         
 [99] munsell_0.5.0               bslib_0.4.0                </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
